@sparques please note test failure:

```
--- FAIL: TestConfigLinkerFlavor (0.00s)
    --- FAIL: TestConfigLinkerFlavor/default_coff (0.00s)
        target_test.go:158: LinkerFlavor() = "gnu", want "coff"
    --- FAIL: TestConfigLinkerFlavor/default_darwin (0.00s)
        target_test.go:158: LinkerFlavor() = "gnu", want "darwin"
FAIL
```

--- a/compileopts/target_test.go
+++ b/compileopts/target_test.go
@@ -150,8 +150,9 @@ func TestConfigLinkerFlavor(t *testing.T) {
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
+			tc.target.GOOS = tc.goos
 			config := &Config{
-				Options: &Options{GOOS: tc.goos},
+				Options: &Options{},
 				Target:  tc.target,
 			}
 			if got := config.LinkerFlavor(); got != tc.want {

